### PR TITLE
Code coverage community data

### PR DESCRIPTION
**Pull request by nmonterroso**
@drsnyder 
https://wikia-inc.atlassian.net/browse/SERVICES-219
Improving quality of tests and increasing code coverage for the community-data service

### CODE REVIEW

File: lib/jooq/jooq-base/src/main/java/com/wikia/test/jooq/MockDataProviderBuilder.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          record.setValue(field, valueList.get(i));
+        }
+
+        result.add(record);
+      }
+
+      return new MockResult[] {new MockResult(valueListSize, result)};
+    };
+  }
+}
```
Comment (drsnyder) [inline lib/jooq/jooq-base/src/main/java/com/wikia/test/jooq/MockDataProviderBuilder.java]: It would be nice to split this function up. It has three loops wrapped in an anonymous function that's being returned. It's pretty hard to follow what is happening.

File: lib/service-core/src/main/java/com/wikia/pandora/http/StandardResponses.java [hunk 2]
```diff
@@ -4,38 +4,27 @@
...
-                    .status(Response.Status.CONFLICT.getStatusCode())
-                    .title(title)
-                    .build())
+    return Response.status(status)
+        .entity(problem)
         .type(Problem.MEDIA_TYPE)
         .build();
   }
-
 }
```
Comment (drsnyder) [inline lib/service-core/src/main/java/com/wikia/pandora/http/StandardResponses.java]: :+1:

**Comment by drsnyder** [comment]
Minor comments. OTW, LGTM

File: lib/jooq/jooq-base/src/main/java/com/wikia/test/jooq/MockDataProviderBuilder.java (hunk 1 above)
Comment (nmonterroso) [inline lib/jooq/jooq-base/src/main/java/com/wikia/test/jooq/MockDataProviderBuilder.java]: 032bb61

File: lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java [hunk 3]
```diff
@@ -4,13 +4,13 @@
 import com.wikia.nirvana.generated.SwaggerApi;
 
 public class NirvanaHelper {
-  public static <T extends SwaggerApi> T getApi(String domain, Class<T> cls)
+  public <T extends SwaggerApi> T getApi(String domain, Class<T> cls)
```
Comment (sqreek) [inline lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java]: why not static? it looks like it should be helper.

Comment (nmonterroso) [inline lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java]: was originally `static`, but Mockito can't mock `static` functions, which is useful here for testing purposes.

Comment (sqreek) [inline lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java]: PowerMock can do this
https://code.google.com/p/powermock/wiki/MockitoUsage

Comment (sqreek) [inline lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java]: Also if you want go non static way -> change NirvanaHelper for something else (Helper give me assumption that all public method inside are static), or
If you go back to static and use Powermockito, add protected constructor to prevent instance creation of this class

Comment (nmonterroso) [inline lib/nirvana-api/src/main/java/com/wikia/nirvana/NirvanaHelper.java]: I don't think it's worth it to introduce a new library for this. I've changed the class name to NirvanaApiProvider in 255ef41
